src/components/tl/nccl/tl_nccl_team.c

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_nccl.h"
+#include "tl_nccl_coll.h"
+
+UCC_CLASS_INIT_FUNC(ucc_tl_nccl_team_t, ucc_base_context_t *tl_context,
+                    const ucc_base_team_params_t *params)
+{
+    ucc_tl_nccl_context_t *ctx    =
+        ucc_derived_of(tl_context, ucc_tl_nccl_context_t);
+    ucc_status_t status;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_tl_team_t, &ctx->super);
+    self->oob       = params->params.oob;
+    self->size      = self->oob.participants;
+    self->rank      = params->rank;
+    self->unique_id = ucc_malloc(sizeof(ncclUniqueId) * (self->size + 1),
+                                 "tl_nccl_unique_id");
+    if (!self->unique_id) {
+        tl_error(ctx->super.super.lib,
+                 "failed to allocate %zd bytes for unique_id array",
+                 sizeof(ncclUniqueId) * (self->size + 1));
+        return UCC_ERR_NO_MEMORY;
+    }
+    if (self->rank == 0) {
+        NCCLCHECK_GOTO(ncclGetUniqueId(&self->unique_id[self->size]),
```

Comment:
this will produce a hang, all ranks except for 0 will hang in the allgather. do we want to use a special value to indicate the error and check after the allgather?